integration/apps/build_func.sh

```diff
@@ -73,9 +74,21 @@ clone_repo_git(){
         return 1
     fi
 
+    backup_archive ${ARCHIVE} ${DIRNAME}
+}
+
+backup_archive() {
+    local ARCHIVE=$1
+    local DIRNAME=$2
+
     # If the user set the cache dir, pack up the source for next time
     if [ -d ${GEOPM_APPS_SOURCES} ]; then
-        tar czvf ${GEOPM_APPS_SOURCES}/${ARCHIVE} ${DIRNAME}
+        if [ ! -f ${ARCHIVE} ]; then
```

Comment:
Can you add some comments about why these changes are required, it is not really clear to me what the effect of these changes is.

Reply:
This one is easier if you use the split view instead of unified.  I refactored this code that was responsible for backing up downloads in GEOPM_APPS_SOURCES.  Previously there were 2 different backup paths, one in clone_repo_git() and one in get_archive().

They were doing almost the same operations, so I did an extract refactor so all the backup code is in the same place.

backup_archive() is just a refactor.  No functional change.